ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/BeaconState.java

```diff
@@ -147,6 +151,201 @@ public BeaconState(
+          // Validator registry
+          writer.writeBytesList(validator_registryBytes);
+          writer.writeULongIntList(
+              64,
+              validator_balances
+                  .stream()
+                  .map(UnsignedLong::longValue)
+                  .collect(Collectors.toList()));
+          writer.writeUInt64(validator_registry_update_epoch.longValue());
+          // Randomness and committees
+          writer.writeBytesList(latest_randao_mixes);
+          writer.writeUInt64(previous_epoch_start_shard.longValue());
+          writer.writeUInt64(current_epoch_start_shard.longValue());
+          writer.writeUInt64(previous_calculation_epoch.longValue());
+          writer.writeUInt64(current_calculation_epoch.longValue());
+          writer.writeBytes(previous_epoch_seed);
+          writer.writeBytes(current_epoch_seed);
+          // Finality
+          writer.writeUInt64(previous_justified_epoch.longValue());
+          writer.writeUInt64(justified_epoch.longValue());
+          writer.writeUInt64(justification_bitfield.longValue());
+          writer.writeUInt64(finalized_epoch.longValue());
+          // Recent state
+          writer.writeBytesList(latest_crosslinksBytes);
+          writer.writeBytesList(latest_block_roots);
+          writer.writeBytesList(latest_index_roots);
+          writer.writeULongIntList(
+              64,
+              latest_penalized_balances
+                  .stream()
+                  .map(UnsignedLong::longValue)
+                  .collect(Collectors.toList()));
+          writer.writeBytesList(latest_attestationBytes);
+          writer.writeBytesList(batched_block_roots);
+          // Ethereum 1.0 chain data
+          writer.writeBytes(latest_eth1_data.toBytes());
+          writer.writeBytesList(eth1_data_votesBytes);
+        });
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(
+        slot,
+        genesis_time,
+        fork,
+        validator_registry,
+        validator_balances,
+        validator_registry_update_epoch,
+        latest_randao_mixes,
+        previous_epoch_start_shard,
+        current_epoch_start_shard,
+        previous_calculation_epoch,
+        current_calculation_epoch,
+        previous_epoch_seed,
+        current_epoch_seed,
+        previous_justified_epoch,
+        justified_epoch,
+        justification_bitfield,
+        finalized_epoch,
+        latest_crosslinks,
+        latest_block_roots,
+        latest_index_roots,
+        latest_penalized_balances,
+        latest_attestations,
+        batched_block_roots,
+        latest_eth1_data,
+        eth1_data_votes);
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (Objects.isNull(obj)) {
+      return false;
+    }
+
+    if (this == obj) {
+      return true;
+    }
+
+    if (!(obj instanceof BeaconState)) {
+      return false;
+    }
+
+    BeaconState other = (BeaconState) obj;
+    return slot.equals(other.getSlot())
```

Comment:
Preference is to use java.util.Objects.equals(obj1, obj2) because it is nullsafe.